staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -119,7 +126,6 @@ func ValidateCustomResourceDefinitionSpecUpdate(spec, oldSpec *apiextensions.Cus
 
 	if established {
 		// these effect the storage and cannot be changed therefore
-		allErrs = append(allErrs, genericvalidation.ValidateImmutableField(spec.Version, oldSpec.Version, fldPath.Child("version"))...)
```

Comment:
Letting this change means we allow storage to contain versions we don't know about.

That implies CR storage doing no-op conversions when reading from etcd, overwriting apiVersion to match the expected version for the handler, right?

We can't limit that the things that match version or transitionalVersion, since you could change version from "v1" to "v2", leaving "v1" data in etcd and no record in the CRD that "v1" was a thing we knew about.

Reply:
> That implies CR storage doing no-op conversions when reading from etcd, overwriting apiVersion to match the expected version for the handler, right?

Yes, we would ignore the version in etcd. 

Which leads to the question what we do when we really have to depend on the version string some day, e.g. for conversions. I am sure one can come up with a solution, but I fear it won't be pretty.